coffee-module/coffee-module-redisstream/src/main/java/hu/icellmobilsoft/coffee/module/redisstream/consumer/RedisStreamConsumerExecutor.java

```diff
@@ -190,20 +216,55 @@ protected void executeProcess(StreamEntry streamEntry, int counter) throws BaseE
      * Process execution wrapper. Running process in self started request scope
      * 
      * @param streamEntry
-     *            Redis stream entry
+     *            Redis stream input entry
+     * @return {@code Optional} result data from {@code IRedisStreamPipeConsumer#onStream(StreamEntry)}
      * @throws BaseException
-     *             exceptin is error
+     *             exception is error
      */
-    protected void executeProcessInRequestScope(StreamEntry streamEntry) throws BaseException {
+    protected Optional<Map<String, Object>> onStreamInRequestScope(StreamEntry streamEntry) throws BaseException {
         // get reference for the consumerBean
-        IRedisStreamConsumer consumer = (IRedisStreamConsumer) beanManager.getReference(consumerBean, consumerBean.getBeanClass(),
-                beanManager.createCreationalContext(consumerBean));
+        Object consumer = beanManager.getReference(consumerBean, consumerBean.getBeanClass(), beanManager.createCreationalContext(consumerBean));
 
         Map<String, Object> requestScopeStore = null;
         try {
             requestScopeStore = new ConcurrentHashMap<>();
             startRequestScope(requestScopeStore);
-            consumer.onStream(streamEntry);
+            if (consumer instanceof IRedisStreamConsumer) {
+                ((IRedisStreamConsumer) consumer).onStream(streamEntry);
+            } else if (consumer instanceof IRedisStreamPipeConsumer) {
+                Map<String, Object> result = ((IRedisStreamPipeConsumer) consumer).onStream(streamEntry);
+                return Optional.of(result);
+            }
+            return Optional.empty();
+        } finally {
+            endRequestScope(requestScopeStore);
+        }
+    }
+
+    /**
+     * Process execution wrapper. Running {@code IRedisStreamPipeConsumer#afterAck(StreamEntry, Map)} process in self started request scope
+     * 
+     * @param streamEntry
+     *            Redis stream input entry
+     * @param onStreamResult
+     *            result of {@code IRedisStreamPipeConsumer#onStream(StreamEntry)}
+     * @throws BaseException
+     *             exception is error
+     */
+    protected void afterAckInRequestScope(StreamEntry streamEntry, Map<String, Object> onStreamResult) throws BaseException {
+        if (!consumerBean.getBeanClass().isAssignableFrom(IRedisStreamPipeConsumer.class)) {
+            return;
```

Comment:
trace szintu log uzenet jol johet

Reply:
rengeteg logot jelentene meg fejlesztesi ido alatt is, minden klasszikus event vegrehajtas logolna egyet...

Reply:
oks, akkor jo igy, koszi a valaszt